src/masternodes/rpc_tokens.cpp

```diff
@@ -815,6 +815,155 @@ UniValue minttokens(const JSONRPCRequest& request) {
+        },
+        RPCExamples{
+            HelpExampleCli("decodecustomtx", "\"hexstring\"")
+            + HelpExampleRpc("decodecustomtx", "\"hexstring\"")
+        },
+    }.Check(request);
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VBOOL});
+
+    CMutableTransaction mtx;
+
+    bool try_witness = request.params[1].isNull() ? true : request.params[1].get_bool();
+    bool try_no_witness = request.params[1].isNull() ? true : !request.params[1].get_bool();
+
+    if (!DecodeHexTx(mtx, request.params[0].get_str(), try_no_witness, try_witness)) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+    }
+
+    int nHeight{0};
+    bool actualHeight{false};
+    CustomTxType guess;
+    UniValue txResults(UniValue::VOBJ);
+    Res res{};
+    uint256 hashBlock;
+    CBlockIndex* blockindex{nullptr};
+    CTransactionRef tx = MakeTransactionRef(std::move(mtx));
+    std::string warnings;
+
+    if (tx)
+    {
+        // Search wallet if available
+        std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+        if (wallet) {
+            LOCK(wallet->cs_wallet);
+            if (auto wtx = wallet->GetWalletTx(tx->GetHash()))
+            {
+                hashBlock = wtx->hashBlock;
+            }
+        }
+
+        // No wallet or not a wallet TX, try mempool, txindex
+        if (!wallet || hashBlock.IsNull())
+        {
+            bool f_txindex_ready{false};
+            if (g_txindex) {
+                f_txindex_ready = g_txindex->BlockUntilSyncedToCurrentChain();
+            }
+
+            //get block hash details
+            if (!GetBlockHash(tx->GetHash(), Params().GetConsensus(), hashBlock, blockindex)) {
+                if (!g_txindex) {
+                    warnings = "Failed to find the relavant block. Use -txindex";
+                } else if (!f_txindex_ready) {
+                    warnings = "Failed to find the relavant block. Transactions are still in the process of being indexed.";
+                } else {
+                    warnings = "Failed to find the relavant block";
+                }
+            }
+        }
+
+        LOCK(cs_main);
+        // Found a block hash but no block index yet
+        if (!hashBlock.IsNull() && !blockindex) {
+            blockindex = LookupBlockIndex(hashBlock);
+        }
+
+        // Default to next block height
+        nHeight = ::ChainActive().Height() + 1;
+
+        // Get actual height if blockindex avaiable
+        if (blockindex) {
+            nHeight = blockindex->nHeight;
+            actualHeight = true;
+        }
+
+        // Skip coinbase TXs except for genesis block
+        if ((tx->IsCoinBase() && nHeight > 0)) {
+            return "Coinbase transaction. Not a custom transaction.";
+        }
+
+        if (!GetCustomTXInfo(nHeight, tx, guess, res, txResults)) {
+            return "Not a custom transaction";
+        }
+
+    } else {
+        // Should not get here
```

Comment:
I think here we don't have to error out, we can still return the metadata, see GuessCustomTxType, we don't need any on-chain info to get info such as the parameters etc, only for the result